pandas/tests/util/test_hashing.py

```diff
@@ -71,6 +71,15 @@ def test_hash_array_errors(val):
         hash_array(val)
 
 
+def test_hash_array_index_exception():
```

Comment:
can you tests other things, e.g. a non-array/extension array (which should raise as well)

Reply:
we have a few others just above this